experimental/displaynames/src/provider.rs

```diff
@@ -0,0 +1,36 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+#![warn(unused_imports)]
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use icu_provider::prelude::*;
+use tinystr::TinyAsciiStr;
+use zerovec::ZeroMap;
+
+// We use raw TinyAsciiStrs for map keys, as we then don't have to
+// validate them as subtags on deserialization. Map lookup can be
+// done even if they are not valid tags (an invalid key will just
+// become inaccessible).
+type UnvalidatedRegion = TinyAsciiStr<14>;
+
+#[icu_provider::data_struct(DisplayNamesV1Marker = "displaynames/territories@1")]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_displaynames::provider),
+)]
+#[yoke(prove_covariance_manually)]
+/// DisplayNames provides mapping between a region code and locale display name.
+pub struct DisplayNamesV1<'data> {
```

Comment:
This should be a region-specific struct named, e.g., `TerritoryDisplayNamesV1`

Reply:
Done